new.go

```diff
@@ -607,27 +609,21 @@ func (s *interim) writeStoredFields() (
 		}
 
 		metaBytes := s.metaBuf.Bytes()
-
-		compressed = snappy.Encode(compressed[:cap(compressed)], data)
-
-		docStoredOffsets[docNum] = uint64(s.w.Count())
-
-		err = writeUvarints(s.w,
-			uint64(len(metaBytes)),
-			uint64(len(compressed)))
-		if err != nil {
-			return 0, err
-		}
-
-		_, err = s.w.Write(metaBytes)
+		docStoredOffsets[docNum] = docChunkCoder.BufferSize()
```

Comment:
In the trunk coder write data document by document, So need log the offset before write a new document.

Reply:
Just doing an initial overview review, and this jumped out at me too.  I think it reads as wrong to me because it feels like we're trying to record some important per-document value, and instead we're recording the size of the buffer,  something that feels static and unrelated.

I sort of get the explanation (but haven't really gone through it closer myself), but I'm still left wondering if we could change a method name, or add another method, to make the code read better (especially since two of found it confusing as is)

Reply:
Okay, i will change the name.